### DISCUSSION TITLE
Cesium Godot?

### DISCUSSION THREAD

**Discussion by GeorgeS2019** [godotengine/godot-proposals#6913] [3D]
https://github.com/CesiumGS/ ==> cesium-godot?
![image](https://github.com/godotengine/godot-proposals/assets/49812372/87cbca2f-72e6-4666-bdd7-b26bc88f3cab)

https://cesium.com/learn/unity/unity-photorealistic-3d-tiles/

![image](https://github.com/godotengine/godot-proposals/assets/49812372/86be3934-26ae-45b0-b2c2-b0fe8c26784d)

**Comment by GeorgeS2019** [comment]
FYI
@BastiaanOlij 



![image](https://github.com/godotengine/godot-proposals/assets/49812372/6f5ccdfd-1a65-4ff2-a912-c8536fe60562)

**Reply by GeorgeS2019** [reply]
Relevant repos
- https://github.com/RodZill4/godot-openstreetmap (last update 2020)
   -  https://yhb37.itch.io/yhbgodotmap
- https://github.com/fcazalet/godot-osm-tilemap
- https://github.com/Geoboardman/Godot_GPS_Base
- https://github.com/Toshiwoz/GodotMapLoader
- https://github.com/rafgro/godot25dmap

**Comment by luxun0517** [comment]
https://youtu.be/WMvIuDC1fUs

**Comment by GeorgeS2019** [comment]
@luxun0517 
@bastiaanOlij
Are U creating Gdextension on top of Cesium native?

Great job for the entire Godot community

**Reply by luxun0517** [reply]
Yes!

**Comment by Khasehemwy** [comment]
@luxun0517 
Could you share a 4.2 or master version of GDExtension zip and sample project (Windows x64)? Thank you! I tried GDExtension in your repositories, but it can only be compiled successfully in the master branch(4.2 will be errors), and I don’t know how to see the effect after creating GD3DTileset😢.

**Reply by luxun0517** [reply]
sorry, my public repositories just for test,  this project is on the way。
you can watch my yutube channel [](https://www.youtube.com/@3DGIS-luxun)

**Reply by luxun0517** [reply]
https://www.youtube.com/@3DGIS-luxun

**Reply by Khasehemwy** [reply]
Okay, okay, thank you again!

**Comment by GeorgeS2019** [comment]
@luxun0517 
@Khasehemwy 
https://github.com/stuarta0/godot-cesium/issues/1

**Comment by Sauermann** [comment]
https://cesium.com/blog/2024/09/26/cesium-plugin-for-godot/

**Reply by GeorgeS2019** [reply]
I need this good news

**Comment by fire** [comment]
https://github.com/Battle-Road-Labs/3D-Tiles-For-Godot

Godot Engine 3D tiles from cesium.

- [x] ~~editor view needs to be rendering~~
- [x] ~~review from @aaronfranke~~
- [x] ~~review from @fire~~
- [x] ???
- [x] portability

**Reply by GeorgeS2019** [reply]
Thx..this is the real official funded Godot caesium

**Reply by aaronfranke** [reply]
Can someone explain the impact of this? What is to be gained from approval by me or other Godot contributors? Is the plan to make this an official add-on? What does funding have to do with this?

**Reply by GeorgeS2019** [reply]
@BastiaanOlij

What is your view of the importance of this officially funded Cesium project specifically for Godot. Especially Godot XR community 

This is finally here after many months of waiting...

How best should the different Godot communities coordinate this official plugin development?

**Reply by BastiaanOlij** [reply]
@GeorgeS2019 this looks like a really cool project, but it's something organised by Cesium so it's up to them to encourage and maintain the plugin. Hopefully this will find an audience within the community and if someone makes something cool with it we can definitely give that some attention.

But its not up to core Godot developers to get involved here unless they have a personal interest in this. As always, FOSS is driven by those who use the software.

**Reply by dsnopek** [reply]
If they want to, they can request to add it to https://github.com/godot-sdk-integrations but that is up to the maintainers

Aside: I just saw their presentation at GodotCon Boston and this looks really cool!

**Reply by BastiaanOlij** [reply]
Really cool they showed it off there, that is excellent!

**Reply by Kyn21kx** [reply]
Hi all, I'm the developer of the 3D tiles for Godot project, I believe the plugin can live on its own as a separate component from Godot maintained by BattleRoadLabs with support from Cesium, although flattering, I don't see the need for this being a core addon of the engine, which, if you really need to be closely integrated into Godot, you can compile the plugin as a Godot submodule along with the rest of the engine's code, that's the beauty of it being written in C++ c:
As for an editor view, it was mostly lack of time that prevented me from implementing that on our first release, but you can expect further development on that.

**Reply by BastiaanOlij** [reply]
Just to clarify @Kyn21kx , the `godot-sdk-integrations` repo doesn't mean an add-on hosted there is core or supported by the foundation. The Foundation created that as a repository where we offer people with popular plugins to host their plugins so they become easier to find and hopefully increase the likelihood that plugins remain supported long after the original author moves on from the project.

That said, seeing your plugin is officially supported by Cesium and hopefully is advertised by Cesium, those users who need the plugin will find it regardless of where it is hosted and can have faith that Cesium keeps supporting the plugin.

**Comment by GeorgeS2019** [comment]
Someone who has right need to re open this discussion

**Reply by BastiaanOlij** [reply]
This isn't the proper place to have this discussion (though its nice that this plugin is now on my radar), discussions belong on the plugin repo itself and/or Cesium support forums.

**Comment by GeorgeS2019** [comment]
I have written at the Cesium forum for godot that we need to start discussing how to port Cesium for Unity tutorials to Godot4. By doing so, we will confident the Cesium for Godot Jas high coverage of API as unity 

https://community.cesium.com/t/porting-unity-cesium-tutorials-to-godot/40630

**Comment by GeorgeS2019** [comment]
[Github Action for Windows build ](https://github.com/Battle-Road-Labs/3D-Tiles-For-Godot/issues/89)for Cesium for Godot is now finally working with help from the Cesium Team

**Comment by GeorgeS2019** [comment]
@BastiaanOlij 

## Sept 2026 Update =>NEW  Cesium for Godot based on **Unrael for Cesium API** 
https://github.com/brynnb/cesium-godot/issues/7#issue-5334149586
